src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -56,35 +56,37 @@ namespace impl {
  */
 template <bool Serialize, typename ValueType, typename TeamMember, typename Lambda>
 static KOKKOS_INLINE_FUNCTION
-void parallel_reduce (const TeamMember& team,
-                      const int& begin,
-                      const int& end,
-                      const Lambda& lambda,
-                      ValueType& result)
+ValueType parallel_reduce (const TeamMember& team,
+                           const int& begin,
+                           const int& end,
+                           const Lambda& lambda,
+                           const ValueType& result)
```

Comment:
I meant that you should consider returning a value without also taking one in: in other words, make this a pure function so it's easy to reason about. Doing both is in some ways even worse that just leaving it as it was.

Reply:
Well, it must take the initial value, in order to do a serialized reduction. I guess I can change the name of the input to avoid confusion. But I do need the input, or else view_reduction would not be bfb in case begin!=0.